tests/end_to_end/end_to_end.cpp

```diff
@@ -341,33 +345,17 @@ droppacket_test(ebpf_execution_type_t execution_type)
     REQUIRE(bpf_map_lookup_elem(dropped_packet_map_fd, &key, &value) == EBPF_SUCCESS);
     REQUIRE(value == 0);
 
+    // Create a normal (not 0-byte) UDP packet.
     packet = prepare_udp_packet(10, ETHERNET_TYPE_IPV4);
-    xdp_md_t ctx2{packet.data(), packet.data() + packet.size()};
+    xdp_md_t ctx2{packet.data(), packet.data() + packet.size(), 0, TEST_IFINDEX};
 
+    // Test that we don't drop the packet.
     REQUIRE(hook.fire(&ctx2, &hook_result) == EBPF_SUCCESS);
-    REQUIRE(hook_result == 1);
+    REQUIRE(hook_result == XDP_PASS);
 
     REQUIRE(bpf_map_lookup_elem(dropped_packet_map_fd, &key, &value) == EBPF_SUCCESS);
     REQUIRE(value == 0);
 
-    fd_t interface_index_map_fd = bpf_object__find_map_fd_by_name(object, "interface_index_map");
```

Comment:
See comment on `droppacket.c`. I think we should keep this test case.

Reply:
Updated